addons/Dimmer/Dimmer.lua

```diff
@@ -53,17 +53,32 @@ function search_item()
     local get_items = windower.ffxi.get_items
     local set_equip = windower.ffxi.set_equip
     
+    -- Get list of all items player has
     for bag_id in pairs(res_bags:equippable(true)) do
         local bag = get_items(bag_id)
         for _,item in ipairs(bag) do
-            if item.id > 0  then
+            if item.id > 0 then
                 item_array[item.id] = item
                 item_array[item.id].bag = bag_id
                 item_array[item.id].bag_enabled = bag.enabled
             end
         end
     end
-    for index,stats in pairs(item_info) do
+
+    -- Narrow list of teleport options
+    local item_options = {}
+    if name then
+      for index,stats in pairs(item_info) do
+        if stats.short_name == name then          
+          table.append(item_options, stats)
```

Comment:
This can be simplified quite a bit. This entire code block could be condensed into one line if you `require('tables')` at the top and prepend the `item_info` table with a `T` like this:
```lua
item_info = T{
    ...
}
```

Then you can write this:
```lua
local item_options = name
    and item_info:filter(function(item) return item.short_name == name end)
    or item_info
```

However, this entire thing asks to be refactored imo. I would probably move the loop contents into its own function `process_item` or whatever, then do this:
```lua
if name then
    local _, stats = item_info:find(function(item) return item.short_name == name end)
    process_item(stats)
else
    for _, stats in pairs(item_options) do
        process_item(stats)
    end
end
```

This could be simplified further if you removed the `short_name` property, and instead just keyed the table by the command:
```lua
item_info = {
    holla={id=26176,japanese='Ｄ．ホラリング',english='"Dim. Ring (Holla)"',slot=13},
    dem  ={id=26177,japanese='Ｄ．デムリング',english='"Dim. Ring (Dem)"',slot=13},
    mea  ={id=26178,japanese='Ｄ．メアリング',english='"Dim. Ring (Mea)"',slot=13},
    mask ={id=10385,japanese="キュムラスマスク+1",english="Cumulus Masque +1",slot=4},
}
```

This can be done because the index is not used anywhere. Now you can simplify it further to:
```lua
if name then
    process_item(item_info[name])
else
    for _, stats in pairs(item_options) do
        process_item(stats)
    end
end
```